experimental/unitsconversion/tests/units_test.rs

```diff
@@ -0,0 +1,155 @@
+        rational_part *= exponent_part;
+        Some(rational_part)
+    }
+
+    #[test]
+    fn test_conversion() {
+        // let provider = icu_unitsconversion::provider::UnitsInfoV1;
+
+        /// Represents a test case for units conversion.
+        #[derive(Debug)]
+        struct UnitsTest {
+            _category: String,
+            input_unit: String,
+            output_unit: String,
+            _result: BigRational,
+        }
+
+        let data = std::fs::read_to_string("tests/data/unitsTest.txt").unwrap();
+        let tests: Vec<UnitsTest> = data
+            .lines()
+            .filter(|line| !line.starts_with('#') && !line.is_empty())
+            .map(|line| {
+                let parts: Vec<&str> = line.split(';').map(|s| s.trim()).collect();
+                UnitsTest {
+                    _category: parts[0].to_string(),
+                    input_unit: parts[1].to_string(),
+                    output_unit: parts[2].to_string(),
+                    _result: get_rational(parts[4]).unwrap(),
+                }
+            })
+            .collect();
+
+        let parser = MeasureUnitParser::from_payload(
+            &icu_unitsconversion::provider::Baked::SINGLETON_UNITS_INFO_V1.units_conversion_trie,
+        );
+
+        // TODO: Those units must be parsable.
+        let non_parsable_units: HashSet<&str> = [
+            "g-force",
+            "arc-second",
+            "arc-minute",
+            "bu-jp",
+            "se-jp",
+            "liter-per-100-kilometer",
+            "mile-per-gallon-imperial",
+            "day-person",
+            "week-person",
+            "pound-force-foot",
+            "calorie-it",
+            "british-thermal-unit",
+            "british-thermal-unit-it",
+            "therm-us",
+            "pound-force",
+            "kilogram-force",
+            "kilowatt-hour-per-100-kilometer",
+            "shaku-length",
+            "shaku-cloth",
+            "jo-jp",
+            "ri-jp",
+            "nautical-mile",
+            "mile-scandinavian",
+            "100-kilometer",
+            "earth-radius",
+            "solar-radius",
+            "astronomical-unit",
+            "light-year",
+            "ounce-troy",
+            "earth-mass",
+            "solar-mass",
+            "solar-luminosity",
+            "pound-force-per-square-inch",
+            "gasoline-energy-density",
+            "dessert-spoon",
+            "dessert-spoon-imperial",
+            "fluid-ounce-imperial",
+            "fluid-ounce",
+            "cup-jp",
+            "cup-metric",
+            "pint-metric",
+            "pint-imperial",
+            "quart-imperial",
+            "gallon-imperial",
+            "to-jp",
+            "month-person",
+            "year-person",
+            "decade",
+        ]
+        .iter()
+        .cloned()
+        .collect();
+        for test in tests {
+            // TODO: remove this line after fixing the parser.
```

Comment:
do you have an issue?

Reply:
done: https://github.com/unicode-org/icu4x/issues/4461